### ISSUE TITLE
Intellisense is flaky with npm modules

### ISSUE THREAD

**Issue by brc-dd** [denoland/deno#24488]
Version: deno 1.44.4+839caf6 (canary, aarch64-apple-darwin)

The intellisense isn't working properly even after #24353. On v1.44.3 and v1.44.4 it was completely broken  (https://github.com/denoland/vscode_deno/issues/1126). But even after that was fixed, it doesn't always work. It seems when the .d.ts of an npm package referencing types from other npm package is opened, the LSP breaks.

https://github.com/denoland/deno/assets/40380293/7f9a85eb-5958-4375-8b10-d17504bc98c2

**Comment by nayeemrmn** [comment]
Closing in favour of #27291, it's more clear.
